packages/contracts/contracts/core/component/ERC165Registry.sol

```diff
@@ -0,0 +1,62 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.8.10;
+
+import "@openzeppelin/contracts/proxy/utils/UUPSUpgradeable.sol";
+import "./Permissions.sol";
+import "../../core/erc165/AdaptiveERC165.sol";
+
+/// @title An ERC165-based registry for contracts
+/// @author Michel Heuer, Sarkawt Noori - Aragon Association - 2022
+/// @notice This contract allows to register contracts
+/// TODO: Find a better nameing instead of ERC165Registry
+abstract contract ERC165Registry is Permissions, UUPSUpgradeable {
```

Comment:
As above, should this `abstract class` have a name like `ERC165RegistryBase` or equivalent?